mlir/utils/performance/analysis/compileAndCollectTuningData.py

```
@@ -0,0 +1,604 @@
+            'WFsPerWG' : self.WFsPerWG,
+            'mfma_wmma_instruction': self.mfma_wmma_instruction
+        }
+
+def get_perf_config(operation, test_vector, arch, num_cu):
+    """
+    Get the performance configuration for the given test vector, architecture,
+    and number of compute units.
+    
+    Args:
+        test_vector: The test vector string.
+        arch: The architecture string.
+        num_cu: The number of compute units.
+    
+    Returns:
+        str: The performance configuration string.
+    """
+    conf_class = perfRunner.PerfConfiguration
+    if (operation == 'attention'):
+        conf_class = perfRunner.AttentionConfiguration.fromCommandLine(test_vector.split(sep=' '), arch, num_cu)
+    elif (operation == 'gemm'):
+        conf_class = perfRunner.GemmConfiguration.fromCommandLine(test_vector.split(sep=' '), arch, num_cu)
+    elif (operation == 'conv'):
+        conf_class = perfRunner.ConvConfiguration.fromCommandLine(test_vector.split(sep=' '), arch, num_cu)
+
+    return conf_class
+
+def compile_config(config, perf_config, operation, paths, timestamp):
+    arch = config[0].split(':')[0]
+    num_cu = config[1]
+    test_vector = config[2]
+
+    conf_class = get_perf_config(operation, test_vector, arch, num_cu)
+    conf_class.setPerfConfig(perf_config)
+    rocmlir_gen_options = conf_class.generateMlirDriverCommandLine("")
+
+    # Build the rocmlir-gen command
+    rocmlir_gen_cmd = [paths.mlir_paths.rocmlir_gen_path] + rocmlir_gen_options.split()
+
+    # Build the rocmlir-driver command
+    rocmlir_driver_cmd = [
+        paths.mlir_paths.rocmlir_driver_path,
+        "-c",
+        f"--arch={arch}",
+        "--debug-only=convert-rock-to-gpu,serialize-to-isa",
+    ]
+
+    commands = [rocmlir_gen_cmd, rocmlir_driver_cmd]
+    out, err = perfRunner.runPipeline(commands)
+
+    # Write debug output to file
+    debug_file = f"rocmlir-driver-debug-{arch}-{timestamp}.txt"
+    with open(debug_file, 'w') as f:
```

Consider adding try-except blocks for working with files (if we intend to run this in CI (to avoid silent errors in ci))